realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmTests.java

```diff
@@ -68,7 +69,18 @@ public void loginLogoutResumeSyncing() throws InterruptedException {
         SyncManager.getSession(config).uploadAllLocalChanges();
         user.logout();
         realm.close();
-        assertTrue(Realm.deleteRealm(config));
+        try {
+            assertTrue(Realm.deleteRealm(config));
+        } catch (IllegalStateException e) {
+            // FIXME: We don't have a way to ensure that the Realm instance on client thread has been
+            //        closed for now.
+            // https://github.com/realm/realm-java/issues/5416
+            if (e.getMessage().contains("It's not allowed to delete the file")) {
+                // retry after 1 second
+                SystemClock.sleep(1000);
```

Comment:
Is it enough to just wait 1 time or should this be done in a loop?

Reply:
will introduce a loop if we observe a failure after this retry